### PR TITLE
Propagate required envs to executors and fix bug to set configs in Spark task

### PR DESCRIPTION
**Pull request by RubenBarragan**
# TL;DR
This PR has two goals: 1) introduce a feature to propagate python configurations to Spark executors, and 2) fix a bug in how Spark configs are set in `SparkContext`/`SparkSession`. In addition, fixed other minor bugs as part of this PR.

## Type
 - [x] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
1. [Feature] Propagate `PYTHONPATH` to executors. If the variable is found in the system, means we already have a setup already prepared, so convey that env to the `SparkSession` to propagate the configuration in executors.
2. [Bug] Fix how configs are set to the `SparkSession`. Changed the way it's set with a `SparkConfig` and introduced it to the 'SparkSession' before creating.

## Tracking Issue
NA

## Follow-up issue
_NA_
OR
_https://github.com/lyft/flyte/issues/<number>_

### CODE REVIEW

**Comment by RubenBarragan** [comment]
@kumare3 (as we synced yesterday) I put together this code initially, I tested this in my use case, and as discussed, it worked pretty well.

Feel free to review and point to better practices/share ideas, please.

**Comment by kumare3** [comment]
So for every commit you do @RubenBarragan you have use
`git commit -s` -> `-s` signs off this is a Linux foundation requirement
Also Can you add a unit test for the pre-execute?

**Comment by kumare3** [comment]
I also see, that we did not add the local_conf, which I think is fine

**Comment by kumare3** [comment]
@RubenBarragan can you add unit test as welll?

**Comment by RubenBarragan** [comment]
Yep, thanks @kumare3 I'm going over the [Contributor Guide](https://docs.flyte.org/projects/flytekit/en/latest/contributing.html) atm.

**Comment by RubenBarragan** [comment]
Yep, thanks @kumare3 I'm going over the [Contributor Guide](https://docs.flyte.org/projects/flytekit/en/latest/contributing.html) atm.

**Comment by wild-endeavor** [comment]
There's a network binding java error in the papermill spark test still, do you know what the issue is there?

**Comment by kumare3** [comment]
@wild-endeavor is it just random failure that spark-driver is unable to get a port on the test runner?

**Comment by wild-endeavor** [comment]
re-running

**Comment by RubenBarragan** [comment]
I couldn't reproduce it locally, so definitely it's happening in CI 🤔

**Review by wild-endeavor — APPROVED** [review]
